samples/subsys/zbus/benchmark/src/benchmark.c

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) 2022 Rodrigo Peixoto <rodrigopex@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "messages.h"
+
+#include <stdint.h>
+
+#include <zephyr/fatal.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/zbus/zbus.h>
+LOG_MODULE_DECLARE(zbus, CONFIG_ZBUS_LOG_LEVEL);
+
+ZBUS_CHAN_DEFINE(bm_channel,		   /* Name */
+		 false,			   /* Read only */
+		 struct external_data_msg, /* Message type */
+
+		 NULL, /* Validator */
+		 ZBUS_OBSERVERS(s1
+
+#if (CONFIG_BM_ONE_TO >= 2LLU)
+				,
+				s2
+#if (CONFIG_BM_ONE_TO > 2LLU)
+				,
+				s3, s4
+#if (CONFIG_BM_ONE_TO > 4LLU)
+				,
+				s5, s6, s7, s8
+#if (CONFIG_BM_ONE_TO > 8LLU)
+				,
+				s9, s10, s11, s12, s13, s14, s15, s16
+#endif
+#endif
+#endif
+#endif
```

Comment:
Something for a followup - let's not add more feature to this - it would be cool to have a way of defining observers throughtout the code with iterable sections instead of an explicit references. That is, a set of macros to declare the memory section with list of observer structures, and then something to define the observer so that it ends up there an gets picked up for the channel without an explicit reference. That would clean up patterns like this, and would allow spreading the subscribers/listeners around multiple source file and enabling/disabling them statically by just excluding the specific source from the build, as it's done with most ohter low level structures in Zephyr.

Again - just something to discuss as a followup.